### DISCUSSION TITLE
What's the difference between on_fit_start and on_train_start in LightningModule?

### DISCUSSION THREAD

**Discussion by marsggbo** [Lightning-AI/pytorch-lightning#8142] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
What's the difference between `on_fit_start` and `on_train_start` hooks in LightningModule?
![image](https://user-images.githubusercontent.com/13477956/123500499-1a8c8f80-d671-11eb-9598-ebfc6f25c809.png)

**Comment by tshu-w** [comment, marked as answer]
I think the document [here](https://pytorch-lightning.readthedocs.io/en/latest/common/lightning_module.html#hooks) has answered your question very well.

**Reply by awaelchli** [reply]
correct, on `fit_start` is just running much earlier than the `train_start`, because in between them there is also the pretraining routine that e.g. sanity checks the val loop.
